src/main/java/io/cdap/plugin/splunk/source/batch/SplunkBatchSourceConfig.java

```diff
@@ -0,0 +1,448 @@
+    if (autoCancel != null) {
+      arguments.put("auto_cancel", autoCancel);
+    }
+    if (!Strings.isNullOrEmpty(earliestTime)) {
+      arguments.put("earliest_time", earliestTime);
+    }
+    if (!Strings.isNullOrEmpty(latestTime)) {
+      arguments.put("latest_time", latestTime);
+    }
+    if (!Strings.isNullOrEmpty(indexedEarliestTime)) {
+      arguments.put("index_earliest", indexedEarliestTime);
+    }
+    if (!Strings.isNullOrEmpty(indexedLatestTime)) {
+      arguments.put("index_latest", indexedLatestTime);
+    }
+    return arguments;
+  }
+
+  /**
+   * Returns query properties required for Splunk results.
+   *
+   * @param offset start index of data
+   * @param count  number of records to return
+   * @return map of arguments
+   */
+  public Map<String, Object> getResultsArguments(Long offset, Long count) {
+    Map<String, Object> arguments = new HashMap<>();
+    arguments.put("output_mode", outputFormat);
+    arguments.put("offset", offset);
+    arguments.put("count", count);
+    return arguments;
+  }
+
+  @Override
+  public void validate(FailureCollector collector) {
+    super.validate(collector);
+    validateConnection(collector);
+    if (!containsMacro(PROPERTY_SEARCH_STRING)
+      && !Strings.isNullOrEmpty(searchString)) {
+      String search = searchString.trim();
+      if (!search.startsWith("search")
+        && !search.startsWith("|")) {
+        collector.addFailure(
+          String.format("Search String '%s' has to start with the 'search' operator " +
+                          "or another generating command.", searchString), null)
+          .withConfigProperty(PROPERTY_SEARCH_STRING);
+      }
+    }
+    if (!containsMacro(PROPERTY_SEARCH_STRING)
+      && !containsMacro(PROPERTY_SEARCH_ID)) {
+      if (Strings.isNullOrEmpty(searchString)
+        && Strings.isNullOrEmpty(searchId)) {
+        collector.addFailure("Search String or Search Id must be specified.", null)
+          .withConfigProperty(PROPERTY_SEARCH_STRING)
+          .withConfigProperty(PROPERTY_SEARCH_ID);
+      }
+    }
+    if (!containsMacro(PROPERTY_EXECUTION_MODE)
+      && !containsMacro(PROPERTY_SEARCH_ID)) {
+      if (ONESHOT_JOB.equals(executionMode)
+        && !Strings.isNullOrEmpty(searchId)) {
+        collector.addFailure("Search Id is not supported in Execution Mode 'Oneshot'.",
+                             null)
+          .withConfigProperty(PROPERTY_EXECUTION_MODE)
+          .withConfigProperty(PROPERTY_SEARCH_ID);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  void validateConnection(FailureCollector collector) {
+    boolean hasPropertyErrors = checkAuthProperties(collector);
+    if (hasPropertyErrors) {
+      return;
+    }
+
+    AuthenticationType authenticationType = getAuthenticationType();
+    try {
+      SplunkSearchClient client = new SplunkSearchClient(this);
+      client.checkConnection();
+    } catch (IOException | RuntimeException e) {
```

Comment:
RuntimeException is too wide in this case. Can it be more specific to connection problems?

Reply:
Splunk.connect doesn't throw any specific exception related to connection problems. It may throw anything starting from NumberFormatException. There are many other possible exceptions.